drivers/sensor/nordic/qdec_nrfx/qdec_nrfx.c

```diff
@@ -173,81 +174,73 @@ static const struct sensor_driver_api qdec_nrfx_driver_api = {
 	.trigger_set  = qdec_nrfx_trigger_set,
 };
 
-#ifdef CONFIG_PM_DEVICE
-static int qdec_nrfx_pm_action(const struct device *dev,
-			       enum pm_device_action action)
+static int qdec_pm_suspend(const struct device *dev)
 {
 	const struct qdec_nrfx_config *config = dev->config;
-	int ret = 0;
+
+	nrfx_qdec_disable(&config->qdec);
+	qdec_nrfx_gpio_ctrl(dev, false);
+
+	return pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+}
+
+static int qdec_pm_resume(const struct device *dev)
+{
+	const struct qdec_nrfx_config *config = dev->config;
+	int ret;
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+
+	qdec_nrfx_gpio_ctrl(dev, true);
+	nrfx_qdec_enable(&config->qdec);
+
+	return 0;
+}
+
+static int qdec_nrfx_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	int ret = -ENOTSUP;
 
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
-		ret = pinctrl_apply_state(config->pcfg,
-					  PINCTRL_STATE_DEFAULT);
-		if (ret < 0) {
-			return ret;
-		}
-		qdec_nrfx_gpio_ctrl(dev, true);
-		nrfx_qdec_enable(&config->qdec);
-		break;
-
-	case PM_DEVICE_ACTION_TURN_OFF:
-		/* device must be uninitialized */
-		nrfx_qdec_uninit(&config->qdec);
-		ret = pinctrl_apply_state(config->pcfg,
-					  PINCTRL_STATE_SLEEP);
-		if (ret < 0) {
-			return ret;
-		}
+		ret = qdec_pm_resume(dev);
 		break;
 
 	case PM_DEVICE_ACTION_SUSPEND:
```

Comment:
Suspend action is only supported if CONFIG_PM_DEVICE=y, so we can save a bit of flash by excluding it
```
#if CONFIG_PM_DEVICE
case PM_DEVICE_ACTION_SUSPEND:
...
#endif
```